Classes/common/HTTP/CDTURLSession.m

```diff
@@ -53,8 +56,28 @@ - (NSURLSessionDataTask *)dataTaskWithRequest:(NSURLRequest *)request
                                                         NSURLResponse *response,
                                                         NSError *error))completionHandler
 {
+    
+    //make the context
+    CDTURLSessionInterceptorContext *ctx = [[CDTURLSessionInterceptorContext alloc ]initWithRequest:[request mutableCopy]];
+    for ( id obj in self.requestInterceptors) {
+        
+        if( [obj conformsToProtocol:@protocol(CDTURLSessionInterceptor)]){
```

Comment:
Just do `respondsToSelector` not sure we get anything from the protocol conformance?